src/processing/BoardState.java

```diff
@@ -0,0 +1,211 @@
+package processing;
+
+import java.util.*;
+import java.io.Serializable;
+import processing.utility.*;
+import processing.boardobject.*;
+
+/**
+* This BoardState class stores two maps which should be serializable for persistence.
+* ClientBoardState and ServerBoardState will store the object of this class in their class.
+*
+* @author Himanshu Jain, Shruti Umat
+* @reviewer Rakesh Kumar, Satchit Desai
+*/
+
+public class BoardState implements Serializable {
+
+	/**
+	 *  To authenticate that the user deserializing this object
+	 */
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * Map from object ID to the Board Object
+	 */
+	private Map <ObjectId, BoardObject> objIdToBoardObject;
+
+	/**
+	 * Map from position on the board to Priority Queue of objects on that position
+	 */
+	private Map <Position, PriorityQueue <PriorityQueueObject>> posToObjects;
+
+	/**
+	 * Constructor to initialize both the maps
+	 */
+	public BoardState() {
+		objIdToBoardObject = new HashMap <ObjectId, BoardObject> ();
+		posToObjects = new HashMap <Position, PriorityQueue <PriorityQueueObject>> ();
```

Comment:
I guess you did not pass the comparator object to the map @himanshu2047

Reply:
It will be mentioned while creating new  priority queue, here I am creating new HashMap, if you go down it is used while inserting object in the priority queue if the queue is not initialised.